File: Programlama Dilllerinin Prensipleri/odev2/KISIM1/RastgeleKisiDeneme/src/rastgelekisideneme/RastgeleKisiDeneme.java
/**
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.ogr.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 20.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden kütüphaneyi test ettiğimiz programın main kısmı
* </p>
*/

package rastgelekisideneme;

import RastgeleKisiUret.RastgeleKisi;
import RastgeleKisiUret.DosyaOkuma;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Scanner;

public class RastgeleKisiDeneme 
{
    public static void main(String[] args)   throws IOException 
    {
        int cikis = 0;

        do
        {  
            System.out.println("1- Rastgele Kisi Uret");
            System.out.println("2- Uretilmis Dosya Kontrol Et");
            System.out.println("3- Cikis");
            
            //kullanıcıdan işlem seçmesi için giriş alıyoruz.
            Scanner oku = new Scanner(System.in);
            String islem = oku.nextLine();

            String tcNo1;
            String telefonNo1;
            String imeiNo1;
            String imeiNoTemp1;

            String tcNo2;
            String imeiNo2;
            String imeiNoTemp2;

            switch(islem)
            {
                case "1":  
                    
                System.out.println("Kac adet kisi ureteceksiniz?");
                Scanner oku2 = new Scanner(System.in);
                int kisiSayisi = Integer.parseInt(oku2.nextLine());
                
                //dosyaya yazmak için kullanıyoruz, üstüne yazmamak için false yapıyoruz. 
                BufferedWriter writer = new BufferedWriter(new FileWriter("Kisiler.txt", false));

                //girilen kisi sayısı kadar kisi olusturuyoruz.
                for(int i = 0; i < kisiSayisi; i++)
                {
                    RastgeleKisi rk = new RastgeleKisi("random_isimler.txt");
                    DosyaOkuma dosya = new DosyaOkuma("Kisiler.txt");

                    for(int j = 0; j < dosya.getSayac(); j++)
                    {
                       //ürettiğimiz veriyi dosyay yazmadan önce dosyada var mı yok mu kontrol etmek için array'e atıyoruz. 
                      String veriler = dosya.VerileriAl()[j];
                      String []kelimeler = veriler.split("\\s+");
                    
                      //dosyada 6 kelimeden farlı sayıda kelime varsa kontrol etmiyoruz.
                        if(kelimeler.length == 6)
                        {
                            tcNo1 = kelimeler[0];
                            telefonNo1 = kelimeler[4];
                            imeiNoTemp1 = kelimeler[5];      
                            imeiNo1= imeiNoTemp1.replaceAll("([(-)])", "");

                            //tc no kontrolu
                            if(tcNo1.equals(rk.getKimlikNo().toString()))
                            {
                             RastgeleKisi rk2 = new RastgeleKisi("random_isimler.txt");
                             rk.setKimlikNo(rk2.oTcNo);
                             j=0; 
                            }

                            //telefon no kontrolu
                            if(telefonNo1.equals(rk.getTelefonNo().toString()))
                            {
                                RastgeleKisi rk3 = new RastgeleKisi("random_isimler.txt");
                                rk.setTelefonNo(rk3.oTelefonNo);
                                j=0; 
                            }

                            //imei no kontrolu
                            if(imeiNo1.equals(rk.oTelefonNo.getImeiNo().toString()))
                            {
                                 RastgeleKisi rk4 = new RastgeleKisi("random_isimler.txt");
                                 rk.oTelefonNo.setImeiNo(rk4.oTelefonNo.oImeiNo);
                                 j=0; 
                            } 
                        }
                    }
                    
                    //verileri dosyaya yazıyoruz.
                    writer.write(rk.getKimlikNo()+" "+rk.getIsim()  +" "+ rk.getSoyisim() + " " + rk.getYas() + " " + rk.getTelefonNo() + " (" + rk.oTelefonNo.getImeiNo() + ")" );
                    writer.write('\n');   
                }
                
                writer.close();
                System.out.println("kisiler oluşturuldu");
                break;

                case "2":    
                    
                RastgeleKisi rk = new RastgeleKisi("random_isimler.txt");     
                DosyaOkuma dosya = new DosyaOkuma("Kisiler.txt");   
                
                System.out.println("T.C. Kimlik Kontrol");

                int gecerliTcSayisi = 0;
                int gecersizTcSayisi = 0;
                
                for(int k = 0; k < dosya.getSayac(); k++)
                {
                    
                    String veriler1 = dosya.VerileriAl()[k];   
                    String []kelimeler1 = veriler1.split("\\s+");   
                     
                    //dosyada 6 kelimeden farlı sayıda kelime varsa kontrol etmiyoruz.
                    if(kelimeler1.length == 6)
                    {
                        tcNo2 = kelimeler1[0];

                        if(rk.oTcNo.kimlikNoKontrolEt(tcNo2)==true)
                        {
                            gecerliTcSayisi++;
                        }

                        else 
                        {
                            gecersizTcSayisi++;
                        }      
                    }
                }
                
                System.out.println("Gecerli : " + gecerliTcSayisi + "\nGecersiz: " + gecersizTcSayisi + "\n");
                
                System.out.println("IMEI Kontrol");
                
                int gecerliImeiSayisi = 0;
                int gecersizImeiSayisi = 0;
                
                for(int l = 0; l < dosya.getSayac(); l++)
                {
                    String veriler2 = dosya.VerileriAl()[l];
                    String []kelimeler2 = veriler2.split("\\s+");
                    
                    //dosyada 6 kelimeden farlı sayıda kelime varsa kontrol etmiyoruz.
                    if(kelimeler2.length == 6)
                    {
                        imeiNoTemp2 = kelimeler2[5];
                        
                        //imeino daki parantezleri kaldırıyoruz.
                        imeiNo2= imeiNoTemp2.replaceAll("([(-)])", "");
                        
                        if(rk.oTelefonNo.oImeiNo.imeiNoKontrolEt(imeiNo2)==true)
                        {
                            gecerliImeiSayisi++;
                        }

                        else 
                        {
                            gecersizImeiSayisi++;
                        }
                    }
                }
                
                System.out.println("Gecerli : " + gecerliImeiSayisi + "\nGecersiz: " + gecersizImeiSayisi + "\n");
                
                break;
                
                case "3":
                    
                cikis = 1;  
                break;
            }
        }while(cikis != 1);
    }  
}

File: Programlama Dilllerinin Prensipleri/odev2/KISIM2/src/RastgeleKisiDeneme.c
/*
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 26.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden programın test kısmı
* </p>
*/

//Gerekli kütüphaneler ekleniyor
#include "RastgeleKisi.h"

int main()
{
	//Kullanıcı çıkış seçeneğini tuşlayıncaya kadar çalışan do-while döngüsü yazılıyor
	int cikis = 0;
    do
    {  
        printf("%s\n","1- Rastgele Kisi Uret");
        printf("%s\n","2- Uretilmis Dosya Kontrol Et");
        printf("%s\n","3- Cikis");
		
		//Kullanıcının seçtiği seçenek alınıyor
        int islem;
		
		scanf("%d", &islem);
		
		//Gerekli değişkenler tanımlanıyor
        char*  tcNo1;
        char*  telefonNo1;
        char*  imeiNo1;
        char*  imeiNoTemp1;

        char*  tcNo2;
        char*  imeiNo2;
        char*  imeiNoTemp2;
	
		
		//İlgili işlemleri yapan switch-case oluşturuluyor
        switch(islem)
        {
            case 1:
			//Kullanıcıdan kaç adet kişi üretmek istediği alınıyor
            printf("%s\n","Kac adet kisi ureteceksiniz?");              
            int kisiSayisi;
			scanf("%d", &kisiSayisi);
			
			//Oluşturulan kişilerin yazılacağı dosya açılıyor
			FILE * yazilacakDosya;

			yazilacakDosya = fopen ("Kisiler.txt","w");
				 
			int i;
			//Kişi sayısı kadar rastgele kişi oluşturuluyor ve dosyaya yazılıyor
            for(i = 0; i < kisiSayisi; i++)
            {
                RastgeleKisi rk = RastgeleKisiOlustur(i + 1);
				fprintf(yazilacakDosya,"%s %s %s %d %s %s%s%s",rk->kisi->kimlikNo->tcNo, rk->kisi->isim, rk->kisi->soyisim, rk->kisi->yas, rk->kisi->telefon->telefonNo ,"(",rk->kisi->telefon->oImeiNo->imeiNo,")");
				fprintf(yazilacakDosya,"\n");
				rk->yoket(rk);
            }
            printf("%s\n","Kisiler olusturuldu.");
            fclose(yazilacakDosya);  
            break;

			case 2:;
			//Dosyadaki TC ve IMEIleri kontrol etmek için gerekli değişkenler tanımlanıyor
            RastgeleKisi rk2 = RastgeleKisiOlustur();
			RastgeleKisi rk3 = RastgeleKisiOlustur(); 			
            Dosya dosya2 = DosyaOlustur("Kisiler.txt");
			Dosya dosya3 = DosyaOlustur("Kisiler.txt");
			
            printf("%s","T.C. Kimlik Kontrol\n");
			int k;
			int gecerliTcSayisi = 0;
			int gecersizTcSayisi = 0;
			//Geçerli ve geçersiz TC kimlik numaralarının sayısını bulan kısım yazılıyor
            for(k = 0; k < dosya2->sayac; k++)
            {        
                char* veriler1 = dosya2->oku(dosya2)[k];           
                char* temp1 =  strtok(veriler1," ");
				int sayac2=0;
						
				while(temp1!=NULL)
				{
					sayac2++;
					temp1=strtok(NULL," ");
				} 
                        
                if(sayac2 == 6)
                {
					char* tcNo2 =  strtok(veriler1," ");   
					
                    if(rk2->kisi->kimlikNo->kimlikNoKontrolEt(tcNo2)==true)
                    {
                        gecerliTcSayisi++;
                    }

                    else 
                    {
                        gecersizTcSayisi++;
                    }
                        
                }							
            } 
			//Geçerli ve geçersiz TC kimlik numaralarının sayısı ekrana yazdırılıyor
            printf("%d   %s\n", gecerliTcSayisi, "Gecerli");
			printf("%d   %s\n\n", gecersizTcSayisi, "Gecersiz");
			dosya2->yoket(dosya2);
			rk2->yoket(rk2);
			
            printf("%s","IMEI Kontrol\n");				
			int l;
			int gecerliImeiSayisi = 0;
			int gecersizImeiSayisi = 0;
			//Geçerli ve geçersiz IMEI numaralarının sayısını bulan kısım yazılıyor
            for(l = 0; l < dosya3->sayac; l++)
            {
                char* veriler2 = dosya3->oku(dosya3)[l];
                char* temp2 =  strtok(veriler2," ");
				int sayac3=0;
						
				while(temp2!=NULL)
				{
					sayac3++;
					temp2=strtok(NULL," ");
							
					if(sayac3 == 5)
					{
						imeiNoTemp2= temp2;
							
					}
				} 

                if(sayac3 == 6)
                {
					char* imeiNo2 = strtok(imeiNoTemp2,"()");

                    if(rk3->kisi->telefon->oImeiNo->imeiNoKontrolEt(imeiNo2)==true)
                    {
                        gecerliImeiSayisi++;
                    }

                    else 
                    {
                        gecersizImeiSayisi++;
                    }
                }			
            }
			//Geçerli ve geçersiz IMEI numaralarının sayısı ekrana yazdırılıyor
			printf("%d   %s\n", gecerliImeiSayisi, "Gecerli");
			printf("%d   %s\n\n", gecersizImeiSayisi, "Gecersiz");	
			dosya3->yoket(dosya3);
			rk3->yoket(rk3);
            break;
			
            case 3:
			//Programdan çıkış yapılıyor
            cikis = 1;  
            break;
			default:
			//Hatalı sayının tuşlanması durumunda ekrana hata mesajı yollanıyor
			printf("%s\n","Hatali secim! Menudeki sayilardan birine basiniz.");
			break;
        }
	  

		
    }while(cikis != 1);	
	return 0;
}